src/cloud-resources/src/crd/materialize.rs

```diff
@@ -128,6 +133,30 @@ pub mod v1alpha1 {
             format!("{}-{}", self.name_unchecked(), suffix)
         }
 
+        pub fn environmentd_scratch_volume_storage_requirement(&self) -> Quantity {
+            self.spec
+                .environmentd_scratch_volume_storage_requirement
+                .clone()
+                .unwrap_or_else(|| {
+                    self.spec
+                        .environmentd_resource_requirements
+                        .as_ref()
+                        .and_then(|requirements| {
+                            requirements
+                                .requests
+                                .as_ref()
+                                .or(requirements.limits.as_ref())
+                        })
+                        // TODO: in cloud, we've been defaulting to twice the
+                        // memory limit, but k8s-openapi doesn't seem to
+                        // provide any way to parse Quantity values, so there
+                        // isn't an easy way to do arithmetic on it
+                        .and_then(|requirements| requirements.get("memory").cloned())
+                        // TODO: is there a better default to use here?
+                        .unwrap_or(Quantity("4096Mi".to_string()))
+                })
+        }
```

Comment:
Maybe we should just default to not using it for environmentd if they haven't specified an amount?

Otherwise, writing a parser is a bit annoying, but not particularly hard.
https://github.com/kubernetes-client/python/blob/master/kubernetes/utils/quantity.py